src/plugins/ai_assistant_management/observability/public/constants.ts

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+export const REACT_QUERY_KEYS = {
```

Comment:
Why are we using react-query?

Reply:
Respectfully, I disagree.

How is something like this:

```
  const { isLoading, isError, isSuccess, isRefetching, data, refetch } = useQuery({
    queryKey: [REACT_QUERY_KEYS.GET_KB_ENTRIES, query],
    queryFn: async ({ signal }) => {
      const response = await http.get<{ entries: KnowledgeBaseEntry[] }>(
        `/internal/management/ai_assistant/observability/kb/entries`,
        {
          query: query
            ? {
                query,
              }
            : {},
          signal,
        }
      );
      return response;
    },
    keepPreviousData: true,
    refetchOnWindowFocus: false,
  });
```

more verbose than say this:

```
    useAbortableAsync(
      ({ signal }) => {
        if (!conversationId) {
          const nextConversation = createNewConversation({
            contexts: chatService?.getContexts() || [],
          });
          setDisplayedMessages(nextConversation.messages);
          return nextConversation;
        }

        return service
          .callApi('GET /internal/observability_ai_assistant/conversation/{conversationId}', {
            signal,
            params: { path: { conversationId } },
          })
          .then((nextConversation) => {
            setDisplayedMessages(nextConversation.messages);
            return nextConversation;
          })
          .catch((error) => {
            setDisplayedMessages([]);
            throw error;
          });
      },
      [conversationId, chatService]
    );
```

Reply:
Also, we do not have to add things in two places just because one plugin uses it. 

To be clear: I am not advocating moving to React Query in the AI Assistant plugin, that ship has sailed.

However, for the lightweight plugin that is the settings plugin I do not see why it should be forbidden to use this very mature and capable library.